polyfills/settings/settings.js

```diff
@@ -80,37 +78,21 @@
 
     this.closed = false;
 
-    function _createAndQueueRequest(data) {
-      var request = new FakeDOMRequest(++_currentRequestId, data);
-      Promise.all([navConnPromise, _lock]).then(values => {
-        // When this is executed the promise should have resolved and thus we
-        // should have this.
-        request.data.lockId = _lockId;
-        values[0].sendObject(request);
-      });
-      return request;
-    }
-
-    this.set = function(settings) {
-      return _createAndQueueRequest({
-        operation: 'set',
-        settings: settings
-      });
-    };
-
-    this.get = function(settings) {
-      return _createAndQueueRequest({
-        operation: 'get',
-        settings: settings
-      });
-    };
+    ['set', 'get'].forEach(method => {
+      this[method] = navConnHelper.methodCall.bind(navConnHelper,
+                                                  {
+                                                    methodName: method,
+                                                    numParams: 1,
+                                                    returnValue: FakeDOMRequest,
+                                                    promise: _lock,
+                                                    field: 'lockId'
+                                                  });
+    });
 
     this.serialize = function() {
       return {
-        id: ++_currentRequestId,
-        data: {
-          operation: 'createLock'
```

Comment:
It actually makes more sense leaving this here (and just not using extraData) since createLock is a intrinsic part of this object which will be here always. Just pass a null (or {}) as extradata